### PR TITLE
CONNECTION_CLOSE is non-ack-eliciting

### LINKED ISSUE #3097: Is CONNECTION_CLOSE ACK-eliciting?
At the moment, we state that _a QUIC packet that contains frames other than ACK and PADDING_ ([section 1.2](https://quicwg.org/base-drafts/draft-ietf-quic-transport.html#rfc.section.1.2)), but I wonder if that is true for CONNECTION_CLOSE frame.

A CONNECTION_CLOSE frame never elicits an acknowledgment. IIUC, it is not meant to be governed by the congestion controller.

Considering the aspects, I think we should add CONNECTION_CLOSE to the list of *non-*ACK-eliciting frames.

### PR DESCRIPTION
**Pull request by kazuho**
fixes #3097

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -167,8 +168,8 @@ of frames contained in a packet affect recovery and congestion control logic:
   performance of the QUIC handshake and use shorter timers for
   acknowledgement.
 
-* Packets that contain only ACK frames do not count toward congestion control
-  limits and are not considered in-flight.
+* Packets that contain only ACK or CONNECTION_CLOSE frames do not count toward
```
Comment (marten-seemann) [inline draft-ietf-quic-recovery.md]: It's not clear if the "only" applies only to ACKs or to both.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -192,8 +192,9 @@ QUIC packet:
 
 Ack-eliciting Packet:
 
-: A QUIC packet that contains frames other than ACK and PADDING. These cause a
-  recipient to send an acknowledgment (see {{sending-acknowledgements}}).
+: A QUIC packet that contains frames other than ACK, PADDING, and
+  CONNECTION_CLOSE. These cause a recipient to send an acknowledgment (see
```
Comment (marten-seemann) [inline draft-ietf-quic-transport.md]: If you bundle a CONNECTION_CLOSE with other frames, it's still not ack ack-eliciting, is it?

File: draft-ietf-quic-recovery.md (hunk 1 above)
Comment (kazuho) [reply draft-ietf-quic-recovery.md]: @ianswett Thank you for the suggestion. Applied in f7eed75f.

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -123,7 +123,8 @@ In-flight:
 
 Ack-eliciting Frames:
 
-: All frames besides ACK or PADDING are considered ack-eliciting.
+: All frames besides ACK, PADDING, or CONNECTION_CLOSE are considered
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md]:
```suggestion
: All frames other than ACK, PADDING, and CONNECTION_CLOSE are considered
```

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -2992,9 +2993,8 @@ valid frames? -->
 ## Generating Acknowledgements {#generating-acks}
 
 Endpoints acknowledge all packets they receive and process. However, only
-ack-eliciting packets (see {{QUIC-RECOVERY}}) trigger the sending of an ACK
-frame.  Packets that are not ack-eliciting are only acknowledged when an ACK
-frame is sent for other reasons.
+ack-eliciting packets trigger the sending of an ACK frame.  Packets that are not
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: Can we fix this here?  I know that this isn't this PR, but I would prefer to say:

```suggestion
ack-eliciting packets cause an ACK frame to be sent within the maximum ack delay.  Packets that are not
```

Or something like that.

File: draft-ietf-quic-transport.md [hunk 5]
```diff
@@ -2992,9 +2993,8 @@ valid frames? -->
 ## Generating Acknowledgements {#generating-acks}
 
 Endpoints acknowledge all packets they receive and process. However, only
-ack-eliciting packets (see {{QUIC-RECOVERY}}) trigger the sending of an ACK
-frame.  Packets that are not ack-eliciting are only acknowledged when an ACK
-frame is sent for other reasons.
+ack-eliciting packets cause an ACK frame to be sent within the maximum ack delay.  Packets that are not
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: Formatting nit: This needs to be a new line to make CircleCI happy

**Pull request merged**